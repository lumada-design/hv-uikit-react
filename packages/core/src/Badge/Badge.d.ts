import * as React from "react";
import { StandardProps } from "@material-ui/core";

export interface HvBadgeProps
  extends StandardProps<React.HTMLAttributes<HTMLDivElement>, HvBadgeClassKey> {
  /**
   * Count is the number of unread notifications.
   * Note count and label are mutually exclusive.
   * count is ignored when label is specified at the same time.
   */
  count?: number;
  /**
   * True if count should be displayed.
   * Note showCount and label are mutually exclusive.
   * showCount is ignored when label is specified at the same time.
   */
  showCount?: boolean;
  /**
   * The maximum number of unread notifications to be displayed
   */
  maxCount?: number;
  /**
   * Custom text to show in place of count.
   * Note showCount and label are mutually exclusive.
   * showCount is ignored when label is specified at the same time.
   */
  label?: string;
  /**
   * Icon which the notification will be attached.
   */
  icon?: React.ReactNode;
  /**
   * Text which the notification will be attached.
   */
  text?: string;
  /**
   * Text variant.
   */
  textVariant?: string;
}

export type HvBadgeClassKey =
  | "root"
  | "badgeContainer"
  | "badge"
  | "badgeIcon"
  | "showCount"
  | "showLabel"
  | "badgeOneDigit";

export default function HvBadge(props: HvBadgeProps): JSX.Element | null;
