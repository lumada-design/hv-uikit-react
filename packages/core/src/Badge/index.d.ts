declare module '@hv/uikit-react-core/dist/Badge' {
  import React from 'react'

  class HvBadge extends React.Component<HvBadgeProps> {}

  export default HvBadge

  export interface HvBadgeProps extends React.HTMLAttributes<HvBadge> {
    /**
     *   A Jss Object used to override or extend the styles applied to the badge.
     */
    classes?: {
      /**
       * Styles applied to the component root class.
       */
      root?: string

      /**
       * Styles applied to the component badge position.
       */
      badgePosition?: string

      /**
       * Styles applied to the component badge.
       */
      badge?: string

      /**
       * Styles applied to the component badge count.
       */
      count?: string

      /**
       * Styles applied to the component badge when count greater than 9.
       */
      badgeTwoDigits?: string
    }

    /**
     * Count is the number of unread notifications
     */
    count?: number

    /**
     * True if count should be displayed.
     */
    showCount?: boolean

    /**
     * The maximum number of unread notifications to be displayed
     */
    maxCount?: number

    /**
     * Icon which the notification will be attached.
     */
    icon?: React.ReactNode

    /**
     * Text which the notification will be attached.
     */
    text?: string

    /**
     * Text variant.
     */
    textVariant?: string
  }
}
