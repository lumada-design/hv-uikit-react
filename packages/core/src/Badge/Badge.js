/*
 * Copyright 2019 Hitachi Vantara Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import Typography from "../Typography";

const Badge = props => {
  const { classes, showCount, count, maxCount, icon, text, textVariant } = props;
  const renderedCount = count > maxCount ? `${maxCount}+` : count;
  const Component = icon || text && <Typography variant={textVariant}>{text}</Typography>;

  const badgeClasses = classNames(classes.badgePosition, {
    [classes.badge]: count > 0,
    [classes.showCount]: showCount,
    [classes.badgeIcon]: icon,
    [classes.badgeOneDigit]: showCount && count <= 9
  });

  return (
    <div className={classes.root}>
      {Component}
      <div className={Component ? classes.badgeContainer : null}>
        <div className={badgeClasses}>{showCount && renderedCount}</div>
      </div>
    </div>
  );
};

Badge.propTypes = {
  /**
   *   A Jss Object used to override or extend the styles applied to the badge.
   */
  classes: PropTypes.shape({
    /**
     * Styles applied to the component root class.
     */
    root: PropTypes.string,
    /**
     * Styles applied to the component badge position.
     */
    badgePosition: PropTypes.string,
    /**
     * Styles applied to the component badge.
     */
    badge: PropTypes.string,
    /**
     * Styles applied to the component badge count.
     */
    count: PropTypes.string,
    /**
     * Styles applied to the component badge when count greater than 9.
     */
    badgeTwoDigits: PropTypes.string
  }).isRequired,
  /**
   * Count is the number of unread notifications
   */
  count: PropTypes.number.isRequired,
  /**
   * True if count should be displayed.
   */
  showCount: PropTypes.bool,
  /**
   * The maximum number of unread notifications to be displayed
   */
  maxCount: PropTypes.number,
  /**
   * Icon which the notification will be attached.
   */
  icon: PropTypes.node,
  /**
   * Text which the notification will be attached.
   */
  text: PropTypes.string,
  /**
   * Text variant.
   */
  textVariant: PropTypes.string
};

Badge.defaultProps = {
  showCount: false,
  maxCount: 99,
  icon: null,
  text: null,
  textVariant: null
};

export default Badge;
