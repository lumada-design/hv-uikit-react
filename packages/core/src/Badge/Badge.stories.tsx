import { useState } from "react";
import { css } from "@emotion/css";
import { Meta, StoryObj } from "@storybook/react";
import {
  HvBadge,
  HvBadgeProps,
  HvButton,
} from "@hitachivantara/uikit-react-core";
import { Alert } from "@hitachivantara/uikit-react-icons";

const meta: Meta<typeof HvBadge> = {
  title: "Components/Badge",
  component: HvBadge,
  decorators: [
    (Story) => (
      <div className={css({ display: "flex", gap: 60 })}>{Story()}</div>
    ),
  ],
};
export default meta;

export const Main: StoryObj<HvBadgeProps> = {
  args: {
    count: 1,
    showCount: true,
    label: "",
    maxCount: 99,
    text: "Messages",
    textVariant: "label",
  },
  argTypes: {
    classes: { control: { disable: true } },
    icon: { control: { disable: true } },
  },
  render: (args) => {
    return <HvBadge {...args} />;
  },
};

export const Multiple: StoryObj<HvBadgeProps> = {
  argTypes: {
    classes: { control: { disable: true } },
  },
  render: () => {
    return (
      <>
        <HvBadge count={1} />
        <HvBadge showCount count={8} />
        <HvBadge showCount count={22} />
        <HvBadge showCount count={100} />
        <HvBadge label="100%" />
      </>
    );
  },
};

export const WithIcon: StoryObj<HvBadgeProps> = {
  argTypes: {
    classes: { control: { disable: true } },
  },
  parameters: {
    docs: {
      description: { story: "Badge sample that uses a custom icon." },
    },
  },
  render: () => {
    return (
      <>
        <HvBadge count={0} icon={<Alert />} />
        <HvBadge count={1} icon={<Alert />} />
        <HvBadge showCount count={8} icon={<Alert />} />
        <HvBadge showCount count={88} icon={<Alert />} />
        <HvBadge showCount count={888} icon={<Alert />} />
        <HvBadge label="100%" icon={<Alert />} />
      </>
    );
  },
};

export const WithText: StoryObj<HvBadgeProps> = {
  argTypes: {
    classes: { control: { disable: true } },
  },
  parameters: {
    docs: {
      description: { story: "Badge sample using only text." },
    },
  },
  render: () => {
    return (
      <>
        <HvBadge count={0} text="Events" textVariant="label" />
        <HvBadge count={1} text="Events" textVariant="label" />
        <HvBadge showCount count={8} text="Events" textVariant="title4" />
        <HvBadge showCount count={88} text="Events" textVariant="title4" />
        <HvBadge showCount count={888} text="Events" textVariant="title3" />
        <HvBadge label="100%" text="Events" textVariant="title2" />
      </>
    );
  },
};

export const WithState: StoryObj<HvBadgeProps> = {
  argTypes: {
    classes: { control: { disable: true } },
  },
  parameters: {
    docs: {
      description: {
        story: "Badge sample using react hooks to set the number of events.",
      },
    },
    eyes: { include: false },
  },
  render: () => {
    const [count, setCount] = useState(1);
    const addCount = () => setCount(count * 2);

    return (
      <>
        <HvButton onClick={addCount}>Double Value</HvButton>
        <HvBadge showCount count={count} text="Events" textVariant="title4" />
      </>
    );
  },
};

export const Accessibility: StoryObj<HvBadgeProps> = {
  argTypes: {
    classes: { control: { disable: true } },
  },
  parameters: {
    docs: {
      description: {
        story: "If you want to specify a custom aria-label, use the role prop.",
      },
    },
    eyes: { include: false },
  },
  render: () => (
    <HvBadge
      showCount
      count={25}
      text="Events"
      textVariant="title4"
      role="status"
      aria-label="25 unread notifications"
    />
  ),
};
