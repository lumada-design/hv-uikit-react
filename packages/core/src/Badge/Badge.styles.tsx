import { theme } from "@hitachivantara/uikit-styles";

import { createClasses } from "../utils/classes";

const labelBaseStyle: React.CSSProperties = {
  ...theme.typography.label,
  padding: "0 5px",
  color: theme.colors.atmo1,
  lineHeight: "16px",
};

export const { staticClasses, useClasses } = createClasses("HvBadge", {
  root: { position: "relative", "&>*": { float: "left" } },
  badgeContainer: { width: 0 },
  badgePosition: {},
  badge: {
    borderRadius: theme.space.xs,
    backgroundColor: theme.colors.secondary,
    float: "left",
    minHeight: "8px",
    minWidth: "8px",
  },
  showCount: { ...labelBaseStyle, wordBreak: "keep-all" },
  showLabel: { ...labelBaseStyle, wordBreak: "keep-all" },
  badgeIcon: { position: "relative", top: "1px", left: "-7px" },
  badgeOneDigit: { padding: 0, width: "16px", textAlign: "center" },
});
