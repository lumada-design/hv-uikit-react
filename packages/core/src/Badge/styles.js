/*
 * Copyright 2019 Hitachi Vantara Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const styles = theme => ({
  root: {
    position: "relative",
    "&>*": { float: "left" }
  },
  badgeContainer: {
    width: 0
  },
  badge: {
    borderRadius: `${theme.hv.spacing.xs}px`,
    backgroundColor: theme.hv.palette.accent.acce1,
    float: "left",
    minHeight: "6px",
    minWidth: "6px"
  },
  badgeIcon: {
    position: "relative",
    top: "1px",
    left: "-7px"
  },
  showCount: {
    ...theme.hv.typography.labelText,
    fontFamily: theme.hv.typography.fontFamily,
    padding: "0 5px",
    maxWidth: 30,
    color: theme.hv.palette.atmosphere.atmo1
  },
  badgeOneDigit: {
    padding: 0,
    width: "16px",
    textAlign: "center"
  },
});

export default styles;
