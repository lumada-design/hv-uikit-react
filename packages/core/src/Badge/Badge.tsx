import { HvTypography, HvTypographyVariants } from "../Typography";
import { useDefaultProps } from "../hooks/useDefaultProps";
import { HvBaseProps } from "../types/generic";
import { ExtractNames } from "../utils/classes";

import { staticClasses, useClasses } from "./Badge.styles";

export { staticClasses as badgeClasses };

export type HvBadgeClasses = ExtractNames<typeof useClasses>;

export interface HvBadgeProps extends HvBaseProps {
  /**
   * Count is the number of unread notifications.
   * Note count and label are mutually exclusive.
   * count is ignored when label is specified at the same time.
   */
  count?: number;
  /**
   * True if count should be displayed.
   * Note showCount and label are mutually exclusive.
   * showCount is ignored when label is specified at the same time.
   */
  showCount?: boolean;
  /** The maximum number of unread notifications to be displayed */
  maxCount?: number;
  /**
   * Custom text to show in place of count.
   * Note showCount and label are mutually exclusive.
   * showCount is ignored when label is specified at the same time.
   */
  label?: string;
  /** Icon which the notification will be attached. */
  icon?: React.ReactNode;
  /** Text which the notification will be attached. */
  text?: string;
  /** Text variant. */
  textVariant?: HvTypographyVariants;
  /** A Jss Object used to override or extend the styles applied to the component. */
  classes?: HvBadgeClasses;
}

/**
 * The badge is a component used to notify the user that something has occurred, in the app context.
 */
export const HvBadge = (props: HvBadgeProps) => {
  const {
    classes: classesProp,
    className,
    showCount = false,
    count = 0,
    maxCount = 99,
    label = null,
    icon = null,
    text = null,
    textVariant = undefined,
    ...others
  } = useDefaultProps("HvBadge", props);

  const { classes, cx } = useClasses(classesProp);

  const renderedCount = count > maxCount ? `${maxCount}+` : count;
  // If label is specified and non-empty, render it.
  // If showCount is specified and count > 0, render the count.
  // Otherwise, render nothing on the badge.
  // (Note count=0 should not be rendered to avoid ghosty 0.)
  const renderedCountOrLabel =
    label || (showCount && count > 0 && renderedCount) || null;
  const Component =
    icon || (text && <HvTypography variant={textVariant}>{text}</HvTypography>);

  return (
    <div className={cx(classes.root, className)} {...others}>
      {Component}
      <div className={Component ? classes.badgeContainer : undefined}>
        <div
          className={cx(classes.badgePosition, {
            [classes.badge]: !!(count > 0 || renderedCountOrLabel),
            [classes.showCount]: !!(!label && renderedCountOrLabel),
            [classes.showLabel]: !!label,
            [classes.badgeIcon]: !!icon,
            [classes.badgeOneDigit]: String(renderedCountOrLabel).length === 1,
          })}
        >
          {renderedCountOrLabel}
        </div>
      </div>
    </div>
  );
};
