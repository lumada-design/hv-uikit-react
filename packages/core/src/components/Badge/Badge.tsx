import { Typography, TypographyVariants } from "..";
import { StyledBadge, StyledContainer, StyledRoot } from "./Badge.styles";

export interface BadgeProps extends BaseProps {
  /**
   * Count is the number of unread notifications.
   * Note count and label are mutually exclusive.
   * count is ignored when label is specified at the same time.
   */
  count?: number;
  /**
   * True if count should be displayed.
   * Note showCount and label are mutually exclusive.
   * showCount is ignored when label is specified at the same time.
   */
  showCount?: boolean;
  /** The maximum number of unread notifications to be displayed */
  maxCount?: number;
  /**
   * Custom text to show in place of count.
   * Note showCount and label are mutually exclusive.
   * showCount is ignored when label is specified at the same time.
   */
  label?: string;
  /** Icon which the notification will be attached. */
  icon?: React.ReactNode;
  /** Text which the notification will be attached. */
  text?: string;
  /** Text variant. */
  textVariant?: TypographyVariants;
  /** A Jss Object used to override or extend the styles applied to the empty state component. */
  classes?: {
    root?: string;
    badgeContainer?: string;
    badgePosition?: string;
  };
}

/**
 * The badge is a component used to notify the user that something has occurred, in the app context.
 */
export const Badge = (props: BadgeProps) => {
  const {
    classes,
    className,
    showCount = false,
    count = 0,
    maxCount = 99,
    label = null,
    icon = null,
    text = null,
    textVariant = undefined,
    ...others
  } = props;

  const renderedCount = count > maxCount ? `${maxCount}+` : count;
  // If label is specified and non-empty, render it.
  // If showCount is specified and count > 0, render the count.
  // Otherwise, render nothing on the badge.
  // (Note count=0 should not be rendered to avoid ghosty 0.)
  const renderedCountOrLabel =
    label || (showCount && count > 0 && renderedCount) || null;
  const Component =
    icon || (text && <Typography variant={textVariant}>{text}</Typography>);

  console.log(renderedCountOrLabel);
  return (
    <StyledRoot
      // @ts-ignore
      aria-label={renderedCountOrLabel?.toString()}
      className={classes?.root}
      {...others}
    >
      {Component}
      <StyledContainer
        component={Component ? true : false}
        className={Component ? classes?.badgeContainer : ""}
      >
        <StyledBadge
          className={classes?.badgePosition}
          badge={!!(count > 0 || renderedCountOrLabel)}
          showCount={!!(!label && renderedCountOrLabel)}
          showLabel={!!label}
          badgeIcon={!!icon}
          badgeOneDigit={String(renderedCountOrLabel).length === 1}
        >
          {renderedCountOrLabel}
        </StyledBadge>
      </StyledContainer>
    </StyledRoot>
  );
};
