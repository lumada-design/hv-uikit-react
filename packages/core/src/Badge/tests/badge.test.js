import React from "react";
import { mount, shallow } from "enzyme";
import { Alert } from "@hitachivantara/uikit-react-icons";
import HvProvider from "../../Provider";
import Typography from "../../Typography";
import Badge from "../index";

describe("Badge ", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <HvProvider>
        <Badge count={0} />
      </HvProvider>
    );
  });

  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(wrapper.find(Badge)).toMatchSnapshot();
  });

  it("should render a small dot when count>0 without showCount", () => {
    wrapper = mount(
      <HvProvider>
        <Badge count={12} />
      </HvProvider>
    );
    const divs = wrapper.find("div");

    expect(divs.at(1).text()).toEqual("");
    expect(divs.find("div.HvBadge-badge")).toHaveLength(1);
    expect(divs.find("div.HvBadge-showCount")).toHaveLength(0);
    expect(divs.find("div.HvBadge-showLabel")).toHaveLength(0);
    expect(divs.find("div.HvBadge-badgeOneDigit")).toHaveLength(0);
  });

  it("should render correctly with showCount", () => {
    wrapper = mount(
      <HvProvider>
        <Badge count={12} showCount />
      </HvProvider>
    );

    const divs = wrapper.find("div");
    expect(divs.at(1).text()).toEqual("12");
    expect(divs.find("div.HvBadge-badge")).toHaveLength(1);
    expect(divs.find("div.HvBadge-showCount")).toHaveLength(1);
    expect(divs.find("div.HvBadge-showLabel")).toHaveLength(0);
    expect(divs.find("div.HvBadge-badgeOneDigit")).toHaveLength(0);
  });

  it("should render correctly with showCount and one-digit count", () => {
    wrapper = mount(
      <HvProvider>
        <Badge count={9} showCount />
      </HvProvider>
    );
    const divs = wrapper.find("div");

    expect(divs.at(1).text()).toEqual("9");
    expect(divs.find("div.HvBadge-badge")).toHaveLength(1);
    expect(divs.find("div.HvBadge-showCount")).toHaveLength(1);
    expect(divs.find("div.HvBadge-showLabel")).toHaveLength(0);
    expect(divs.find("div.HvBadge-badgeOneDigit")).toHaveLength(1);
  });

  it("should render nothing when count is 0 even with showCount", () => {
    wrapper = mount(
      <HvProvider>
        <Badge count={0} showCount />
      </HvProvider>
    );
    const divs = wrapper.find("div");

    expect(divs.at(1).text()).toEqual("");
    expect(divs.find("div.HvBadge-badge").length).toBe(0);
    expect(divs.find("div.HvBadge-showCount").length).toBe(0);
    expect(divs.find("div.HvBadge-showLabel").length).toBe(0);
    expect(divs.find("div.HvBadge-badgeOneDigit")).toHaveLength(0);
  });

  it("should render correctly with maxCount", () => {
    wrapper = mount(
      <HvProvider>
        <Badge count={100} showCount />
      </HvProvider>
    );

    const divs = wrapper.find("div");
    expect(divs.at(1).text()).toEqual("99+");
  });

  it("should render correctly with text", () => {
    wrapper = mount(
      <HvProvider>
        <Badge count={100} showCount text="hello" textVariant="sTitle" />
      </HvProvider>
    );

    const text = wrapper.find(Typography);
    expect(text.length).toEqual(1);

    const divs = wrapper.find("div");
    expect(divs.at(1).text()).toEqual("99+");
  });

  it("should render correctly with svg", () => {
    wrapper = mount(
      <HvProvider>
        <Badge count={100} showCount icon={<Alert />} />
      </HvProvider>
    );

    const icon = wrapper.find(Alert);
    expect(icon.length).toEqual(1);

    const divs = wrapper.find("div");
    expect(divs.at(2).text()).toEqual("99+");
  });

  it("should render correctly with custom label", () => {
    wrapper = mount(
      <HvProvider>
        <Badge label="New!" />
      </HvProvider>
    );
    const divs = wrapper.find("div");

    expect(divs.at(1).text()).toEqual("New!");
    expect(divs.find("div.HvBadge-badge")).toHaveLength(1);
    expect(divs.find("div.HvBadge-showCount")).toHaveLength(0);
    expect(divs.find("div.HvBadge-showLabel")).toHaveLength(1);
    expect(divs.find("div.HvBadge-badgeOneDigit")).toHaveLength(0);
  });

  it("should render correctly with custom one-character label", () => {
    wrapper = mount(
      <HvProvider>
        <Badge label="!" />
      </HvProvider>
    );
    const divs = wrapper.find("div");

    expect(divs.at(1).text()).toEqual("!");
    expect(divs.find("div.HvBadge-badge")).toHaveLength(1);
    expect(divs.find("div.HvBadge-showCount")).toHaveLength(0);
    expect(divs.find("div.HvBadge-showLabel")).toHaveLength(1);
    expect(divs.find("div.HvBadge-badgeOneDigit")).toHaveLength(1);
  });

  it("should render custom label but not count when both are specified", () => {
    wrapper = mount(
      <HvProvider>
        <Badge label="New!" count={23} showCount />
      </HvProvider>
    );
    const divs = wrapper.find("div");

    expect(divs.at(1).text()).toEqual("New!"); // not to be "23"
    expect(divs.find("div.HvBadge-badge")).toHaveLength(1);
    expect(divs.find("div.HvBadge-showCount")).toHaveLength(0); // not to exist
    expect(divs.find("div.HvBadge-showLabel")).toHaveLength(1);
    expect(divs.find("div.HvBadge-badgeOneDigit")).toHaveLength(0);
  });
});
