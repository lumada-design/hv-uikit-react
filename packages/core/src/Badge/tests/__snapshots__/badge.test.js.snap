// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge  should render correctly 1`] = `
<JssProvider
  generateClassName={[Function]}
>
  <HvProvider
    changeTheme={[Function]}
    router={null}
    theme={null}
    uiKitTheme="dawn"
  >
    <WithStyles(Badge)
      count={0}
    />
  </HvProvider>
</JssProvider>
`;
